atlasdb-config/src/main/java/com/palantir/atlasdb/http/AtlasDbHttpClients.java

```diff
@@ -100,6 +102,7 @@ private AtlasDbHttpClients() {
      */
     private static Client newOkHttpClient(Optional<SSLSocketFactory> sslSocketFactory) {
         com.squareup.okhttp.OkHttpClient client = new com.squareup.okhttp.OkHttpClient();
+        client.setConnectionPool(new ConnectionPool(100, TimeUnit.MILLISECONDS.convert(10, TimeUnit.MINUTES)));
```

Comment:
Good question.

For now I'd stick it at the top of the class - we can always move it out if we need to use it elsewhere.

Also, can we add a test? Even a simple one that ensures the truth of "`newOkHttpClient` returns a `Client` which can handle `MAX_IDLE_CONNECTIONS`" idle connections would be good.